File: TSDW/Esercitazioni/itinere3/client_order_s/src/main/java/com/example/client_order_s/controllers/OrderController.java
package com.example.client_order_s.controllers;

import org.springframework.stereotype.*;
import org.springframework.web.bind.annotation.*;
import org.springframework.ui.*;

import com.example.client_order_s.models.Order;
import com.example.client_order_s.repositories.ClientRepository;
import com.example.client_order_s.repositories.OrderRepository;

@Controller
public class OrderController {
    
    private ClientRepository repoClient;
    private OrderRepository repo;

    public OrderController(ClientRepository repoClient, OrderRepository repo){
        this.repo=repo;
        this.repoClient=repoClient;
    }

    @GetMapping("/orders")
    public String index(Model model){
        model.addAttribute("orders", repo.findAll());
        return "orders/index";
    }

    @GetMapping("/orders/show")
    public String show(Model model, Long id){
        model.addAttribute("order", repo.findById(id).orElse(null));
        model.addAttribute("client", repoClient.findAll());
        return "orders/show";
    }

    @GetMapping("/orders/create")
    public String crete(Model model){
        model.addAttribute("clients", repoClient.findAll());
        return "orders/create";
    }

    @GetMapping("/orders/edit")
    String edit(Long id, Model model){
        model.addAttribute("order", repo.findById(id).orElse(null));
        model.addAttribute("clients", repoClient.findAll());
        return "/orders/edit";
    }

    @PostMapping("/orders/store")
    public String store(Order order){
        repo.save(order);
        return "redirect:/orders";
    }

    @PostMapping("/orders/update")
    public String update(Order order){
        repo.save(order);
        return "redirect:/orders";
    }

    @PostMapping("/orders/delete")
    public String delete(Long id){
        repo.deleteById(id);
        return "redirect:/orders";
    }
    
}

File: TSDW/Esercitazioni/itinere3/client_order_l/app/Http/Controllers/OrderController.php
<?php

namespace App\Http\Controllers;

use App\Models\Order;
use App\Models\Client;
use Illuminate\Http\Request;

class OrderController extends Controller
{
        /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view('orders.index', ["orders"=>Order::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view('orders.create', ["clients"=>Client::all()]);
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Order::create($request->all());
        return redirect()->route("orders.index");
    }

    /**
     * Display the specified resource.
     */
    public function show(Order $order)
    {
        return view('orders.show', ["order"=>$order]);
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Order $order)
    {
        return view("orders.edit", ["order"=>$order, "client"=>Client::all()]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Order $order)
    {
        $order->update($request->all());
        return redirect()->route("orders.index");
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Order $order)
    {
        $order->delete();
        return redirect()->route('orders.index');
    }
}
